hercules-ci-api/src/Hercules/API/Projects/Job.hs

```diff
@@ -36,3 +37,10 @@ data JobStatus
   | Failure
   | Success
   deriving (Generic, Show, Eq, ToJSON, FromJSON, ToSchema)
+
+data Contextual context a = Contextual
```

Comment:
I think it's better to go with `{ project = ..., job = ...}` as it's clear in the response what's what. We don't expect those two attributes to mean anything else in this particular response, so why make it abstract?

In Haskell I'm fine using `Contextual` abstraction, as long as JSON representation is concrete.